lib/compress.js

```diff
@@ -1731,6 +1731,7 @@ merge(Compressor.prototype, {
                 var val = {};
                 for (var i = 0, len = this.properties.length; i < len; i++) {
                     var prop = this.properties[i];
+                    if (prop instanceof AST_Expansion) return this;
```

Comment:
I'd say changing this condition:
https://github.com/mishoo/UglifyJS2/blob/4700c14855e0189804e0834bc1a12e39c4463961/lib/compress.js#L1430

Now if it turns out restricting to just `AST_ObjectKeyVal` makes most sense, this can be changed on `master` to keep diff between branches minimal.